CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Remoting.Metadata.W3cXsd2001;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CSharp_Net_module1_1_4_lab
+{
+    class Program
+    {
+        // 1) declare enum ComputerType
+
+        enum ComputerType
+        {
+            Desktop,
+            Laptop,
+            Server
+        }
+
+        // 2) declare struct Computer
+
+        struct Computer
+        {
+            public ComputerType type;
+            public int count;
+            public int CPUCores;
+            public float CPUFrequency;
+            public int memory;
+            public int storage;
+        }
+
+        static void Main(string[] args)
+        {
+            // 3) declare jagged array of computers size 4 (4 departments)
+
+            Computer[][] departments = new Computer[4][];
+
+            // 4) set the size of every array in jagged array (number of computers)
+
+            for (int i = 0; i < departments.GetLength(0); i++)
+            {
+                departments[i] = new Computer[3];
+            }
+
+            // 5) initialize array
+            // Note: use loops and if-else statements
+
+            for (int i = 0; i < departments.GetLength(0); i++)
```

Comment:
you shouldn't use an array in that way. Basically, you define each cycle in a different way.
Try to think how to handle it in anotherway.

Reply:
> you shouldn't use an array in that way. Basically, you define each cycle in a different way. Try to think how to handle it in anotherway.

I recreated array to jagged but I don't know how to insert data to a jagged array properly with loops and conditions because its not multidimensional array and indexes are not matching, it will cause out of range error. For example in 2 department there is no desktop computers but you will need to set laptop to index 0 in array. Can you give a hint?

Reply:
You can have a setting matrix.
Rows can be department index and columns can be number or server/desktops/laptops that user have
```
s  l  d
0 2 0 # 2 laptops only
1 2 3 # 1 server, 2 laptops, 3 desktops
```